--- a/src/content/renderer/BUILD.gn
+++ b/src/content/renderer/BUILD.gn
@@ -4,6 +4,7 @@
 
 import("//build/config/buildflags_paint_preview.gni")
 import("//build/config/features.gni")
+import("//build/config/dryft/dryft.gni")
 import("//build/config/ui.gni")
 import("//build/config/compiler/compiler.gni")
 import("//components/nacl/features.gni")
@@ -300,6 +301,11 @@ target(link_target_type, "renderer") {
     ]
   }
 
+  if (enable_nostr) {
+    sources += [ "nostr/nostr_injection.cc", "nostr/nostr_injection.h" ]
+    deps += [ "//components/nostr:nostr_renderer" ]
+  }
+
   if (is_linux || is_chromeos) {
     deps += [ "//components/services/font:lib" ]
   }