--- a/src/chrome/browser/BUILD.gn
+++ b/src/chrome/browser/BUILD.gn
@@ -27,6 +27,7 @@ import("//chrome/common/features.gni")
 import("//chrome/process_version_rc_template.gni")
 import("//components/nacl/features.gni")
 import("//components/optimization_guide/features.gni")
+import("//components/nostr/BUILD.gn")
 import("//extensions/buildflags/buildflags.gni")
 import("//media/media_options.gni")
 import("//ppapi/buildflags/buildflags.gni")
@@ -5200,6 +5201,15 @@ static_library("browser") {
     deps += [ "//chrome/browser/screen_ai:visual_annotations_controller" ]
   }
 
+  if (enable_nostr) {
+    deps += [
+      "//chrome/browser/nostr",
+      "//chrome/browser/blossom",
+      "//chrome/browser/nsite",
+      "//components/nostr",
+    ]
+  }
+
   if (is_chromeos_ash) {
     assert(!is_android)